stdlib.sh

```diff
@@ -986,6 +986,37 @@ layout_pipenv() {
   export VIRTUAL_ENV
 }
 
+# Usage: layout poetry
+#
+# Similar to layout_python, but uses Poetry [1] to build a virtualenv from the
+# pyproject.toml [2] located in the same directory.
+#
+# Poetry by default will create a virtual environment under
+# {cache-dir}/virtualenvs where cache-dir equates to $XDG_CACHE_HOME/pypoetry
+# or use the {project-dir}/.venv directory when one is available.
+#
+# [1]: https://python-poetry.org/
+# [2]: https://peps.python.org/pep-0518/
+#
+layout_poetry() {
+    PYPROJECT_TOML="${PYPROJECT_TOML:-pyproject.toml}"
+    if [[ ! -f "$PYPROJECT_TOML" ]]; then
+        log_error "No pyproject.toml found. Executing \`poetry init\` to create a \`$PYPROJECT_TOML\` first."
```

Comment:
Am I right in assuming there is a [log_msg](https://github.com/direnv/direnv/blob/1be1b52b675916d901780a7905dd5481e4df4864/internal/cmd/log.go#L56)? If so it may be preferable here than `log_error`.